core/trino-main/src/main/java/io/trino/execution/resourcegroups/InternalResourceGroup.java

```diff
@@ -586,6 +589,17 @@ public InternalResourceGroup getOrCreateSubGroup(String name)
         }
     }
 
+    public boolean isKillQuery(Optional<Statement> statement)
```

Comment:
It effectively archives the goal, but also introduces tight coupling between previously unrelated parts of the system (resource groups system, and one of the way to kill a query). We should try to identify more higher level abstraction that would address the use-case.